lib/activegraphql/fetcher.rb

```diff
@@ -0,0 +1,28 @@
+require 'activegraphql/support/fancy'
+
+module ActiveGraphQL
+  class Fetcher < Support::Fancy
+    attr_accessor :url, :klass, :action, :params, :query
+
+    class Error < StandardError; end
+
+    def initialize(attrs)
+      super(attrs)
+      self.query = Query.new(url: url, action: action, params: params)
+    end
+
+    def fetch(*graph)
+      response = query.get(*graph)
+      return if response.blank?
+
+      case response
+      when Hash
+        klass.new(response)
+      when Array
+        response.map { |h| klass.new(h) }
+      else
+        fail Error, "Unexpected response for query: #{response}"
```

Comment:
you still have a fail here.

I think the overall idea is to be consistent. If in one place you're using raise and in another fail then is confusing, same happens for using then after when... or not.